### PR TITLE
EDU-1355 Add basic file upload plugin and capabilities

### PR DESCRIPTION
**Pull request by ahelmberger**
Closes https://educandu.atlassian.net/browse/EDU-1355

### CODE REVIEW

File: src/test-helper.js [hunk 1]
```diff
@@ -156,16 +157,19 @@ export async function updateTestUser(container, user) {
...
     slug: roomValues.slug || '',
     name: roomValues.name || 'my-room',
     shortDescription: roomValues.shortDescription || '',
     isCollaborative: roomValues.isCollaborative || false,
-    ownedBy: roomValues.ownedBy || uniqueId.create(),
-    createdBy: roomValues.createdBy || uniqueId.create(),
-    createdOn: roomValues.createdOn || new Date(),
-    updatedOn: roomValues.updatedOn || new Date(),
+    ownedBy: roomValues.ownedBy || creatorAndOwner,
+    createdBy: roomValues.createdBy || creatorAndOwner,
```
Comment (adrianaluchian) [inline src/test-helper.js:170]: Yet another point in which we anticipated a direction that in the end never came.

File: src/ui/default-prop-types.js [hunk 2]
```diff
@@ -281,10 +281,23 @@ export const documentInputSectionCommentShape = PropTypes.shape({
...
   data: PropTypes.object,
-  comments: PropTypes.arrayOf(documentInputSectionCommentShape)
+  files: PropTypes.arrayOf(PropTypes.shape({
+    key: PropTypes.string.isRequired,
+    name: PropTypes.string.isRequired,
+    size: PropTypes.number.isRequired,
+    type: PropTypes.string.isRequired,
+    url: PropTypes.string.isRequired
+  })).isRequired,
+  comments: PropTypes.arrayOf(PropTypes.shape({
```
Comment (adrianaluchian) [inline src/ui/default-prop-types.js:291]: Here the setup from before needs to stay (`comments: PropTypes.arrayOf(documentInputSectionCommentShape` ), I imagine there was a merge conflict.

File: src/utils/storage-utils.js [hunk 3]
```diff
@@ -21,6 +21,10 @@ export function getRoomMediaRoomPath(roomId) {
   return `room-media/${roomId}`;
 }
 
+export function getDocumentInputPath(roomId, documentInputId) {
```
Comment (adrianaluchian) [inline src/utils/storage-utils.js:24]: `getDocumentInputMediaPath`

File: src/test-helper.js (hunk 1 above)
Comment (ahelmberger) [reply src/test-helper.js:170]: I had the same thought :-)

File: src/plugins/file-upload-field/file-upload-field-display.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+          <Upload
+            multiple
+            listType="picture"
+            maxCount={maxCount}
+            fileList={fileList}
+            onChange={handleChange}
+            disabled={!canModifyInput}
+            beforeUpload={handleBeforeUpload}
+            >
+            {fileList.length >= 8 ? null : uploadButton}
```
Comment (adrianaluchian) [inline src/plugins/file-upload-field/file-upload-field-display.js:54]: Where is this 8 coming from? Should it be from the plugin content `maxSize`?

Comment (ahelmberger) [reply src/plugins/file-upload-field/file-upload-field-display.js]: Yes, well spotted, I fixed it.

File: src/utils/storage-utils.js (hunk 3 above)
Comment (ahelmberger) [reply src/utils/storage-utils.js]: Thanks, corrected!

File: src/ui/default-prop-types.js (hunk 2 above)
Comment (ahelmberger) [reply src/ui/default-prop-types.js]: Actually I inlined it, but reverted now and also extracted the shape for the files array.

File: src/services/document-input-service.js [hunk 5]
```diff
@@ -187,32 +187,34 @@ class DocumentInputService {
         sections: finalSections
       };
 
-      lock = await this.lockStore.takeUserLock(room.ownedBy);
+      if (newCdnFiles.length) {
```
Comment (ahelmberger) [inline src/services/document-input-service.js:190]: I also fixed this in the last commit to only check storage quota if there are actually files in the input.

**Review by adrianaluchian — APPROVED** [review]

**Pull request merged**